platform-tests/src/test/java/org/junit/platform/engine/support/hierarchical/ForkJoinPoolHierarchicalTestExecutorServiceTests.java

```diff
@@ -67,44 +108,128 @@ void defersTasksWithIncompatibleLocks() throws Exception {
 			deferred.countDown();
 		};
 
-		var isolatedTask = new DummyTestTask("isolatedTask", globalReadWriteLock,
-			t -> threadNamesByTaskIdentifier.put(t.identifier(), Thread.currentThread().getName()));
+		var incompatibleTask = new DummyTestTask("incompatibleTask", incompatibleLock);
+
+		var tasks = runWithAttemptedWorkStealing(taskEventListener, incompatibleTask, initialLock, () -> {
+			try {
+				deferred.await();
+			}
+			catch (InterruptedException e) {
+				System.out.println("Interrupted while waiting for task to be deferred");
+			}
+		});
+
+		assertEquals(incompatibleTask, deferredTask.get());
+		assertEquals(tasks.get("nestedTask").threadName, tasks.get("leafTask2").threadName);
+		assertNotEquals(tasks.get("leafTask1").threadName, tasks.get("leafTask2").threadName);
+	}
+
+	static List<Arguments> compatibleLockCombinations() {
+		return List.of(//
+			arguments(//
+				Set.of(GLOBAL_READ), //
+				Set.of(new ExclusiveResource("a", LockMode.READ)) //
+			), //
+			arguments(//
+				Set.of(GLOBAL_READ), //
+				Set.of(GLOBAL_READ, new ExclusiveResource("a", LockMode.READ)) //
+			), //
+			arguments(//
+				Set.of(GLOBAL_READ, new ExclusiveResource("a", LockMode.READ)), //
+				Set.of(GLOBAL_READ, new ExclusiveResource("a", LockMode.READ)) //
+			), //
+			arguments(//
+				Set.of(GLOBAL_READ, new ExclusiveResource("a", LockMode.READ)), //
+				Set.of(GLOBAL_READ, new ExclusiveResource("b", LockMode.READ)) //
+			)//
```

Comment:
Shouldn't you also test some with more than a single additional resource?

Reply:
Done in 5a88d210b7ad79bcd9935e5179fa97da50b16a31.